core/app/service/backup.go

```diff
@@ -107,13 +107,14 @@ func (u *BackupService) GetLocalDir() (string, error) {
 }
 
 func (u *BackupService) SearchWithPage(req dto.SearchPageWithType) (int64, interface{}, error) {
-	count, accounts, err := backupRepo.Page(
-		req.Page,
-		req.PageSize,
-		repo.WithByType(req.Type),
-		repo.WithByName(req.Info),
-		repo.WithOrderBy("created_at desc"),
-	)
+	options := []global.DBOption{repo.WithOrderBy("created_at desc")}
+	if len(req.Type) != 0 {
+		options = append(options, repo.WithByType(req.Type))
+	}
+	if len(req.Info) != 0 {
+		options = append(options, repo.WithByType(req.Info))
+	}
+	count, accounts, err := backupRepo.Page(req.Page, req.PageSize, options...)
 	if err != nil {
 		return 0, nil, err
 	}
```

Comment:
There is no need to optimize the given code as it seems correct and efficient already. The functions named getLocalDir() can also be renamed to avoid conflicts with other function names, such as `SearchWithPage()`.

The comments at lines 50 and 107 seem redundant because they repeat information within those lines; however, these redundancies do not cause errors in the existing code but serve a stylistic purpose rather than improving efficiency. If you find them overly verbose and wish to reduce redundancy without altering functionality, consider updating the variable name, comment header text slightly, or commenting out the additional line entirely:

```
func (u *BackupService) getLocalDir()
   = func (*BackupService) (localDir string, error error){ } +  
   = func (&BackupService)(localDir string, error error){ return }

if err := u.getLocalDir(); err != nil { ... }
```